extension/js/common/settings.ts

```diff
@@ -141,6 +141,7 @@ export class Settings {
     for (const ki of destAccountPrivateKeys) {
       await KeyStore.add(newAcctEmail, ki.private);
     }
+    // todo:
```

Comment:
should we leave this copying as it is or rely on OrgRules?

Reply:
If enforcement already happened in html, then throw if you encounter unexpected state here based on orgrules

Reply:
Well noticed!

```
    // todo:
    for (const fingerprint of Object.keys(destAcctPassPhrases)) {
      await PassphraseStore.set('local', newAcctEmail, fingerprint, destAcctPassPhrases[fingerprint]);
    }
```

This code above was a bug. If the user was unlucky and exactly did email change whenever pass phrase was still memorized in session, then this user's pass phrase would now be stored in hard storage.

We should:
 - fix this bug - if pass phrase was originally in session, then it can be just forgotten. If it was in hard storage, then store it that way again, except:
 -  apply the old user's OrgRules (it's easier) to not store pass phrase in local storage in case OrgRules prevent it, regardless where the pass phrase was originally

You can figure out if pas phrase is in session or local storage by testing with `ignoreSession=true` vs without.

Once the user's change is done, they are now on a new email address, and when they load settings page again, new OrgRules will get loaded. It's this code: `checkFcAcctAndSubscriptionAndContactPage` (however I just saw another bug here, which I'll file separately).

This brings up an interesting question - change of org rules over time. What if I stored my pass phrase in storage, and then OrgRules changed to prevent it later? We will not try to actively address this scenario yet, but I'll file an issue so that we can track it for later.

Reply:
> You can figure out if pas phrase is in session or local storage by testing with `ignoreSession=true` vs without.

This is already done on line 117
```
    for (const ki of destAccountPrivateKeys) {
      const pp = await PassphraseStore.get(newAcctEmail, ki.fingerprints[0], true);
      if (pp) {
        destAcctPassPhrases[ki.fingerprints[0]] = pp;
      }
    }
```
Did you mean the bulk copying on line 133?
```
    await AcctStore.set(newAcctEmail, oldAcctStorage);
```

Reply:
Also, I can't see the point of applying old OrgRules if the domain name has changed.
Let's wipe all passphrases from the storage if you think it's not appropriate to load new OrgRules here?

Reply:
I wonder if new OrgRules were already loaded automatically. OrgRules are loaded immediatelly following Google authentication, with the google OIDC. Maybe if you try loading OrgRules by new name, they will already be saved to storage by now? (I suspect yes but would have to review code further to confirm)